backend/eventloop/shared_resource_loop/sharedresourceloop_repositorycredential.go

```diff
@@ -322,6 +338,37 @@ func internalProcessMessage_ReconcileRepositoryCredential(ctx context.Context,
 				return nil, err
 			}
 
+			normalizedRepoURL := NormalizeGitURL(dbRepoCred.PrivateURL)
+
+			// Check whether the AppProjectRepository exists. If the RepositoryCredentials have been updated and the AppProjectRepository is not present, create it.
+			appProjectRepository := db.AppProjectRepository{
+				Clusteruser_id: clusterUser.Clusteruser_id,
+				RepoURL:        normalizedRepoURL,
+			}
+
+			if err := dbQueries.GetAppProjectRepositoryByUniqueConstraint(ctx, &appProjectRepository); err != nil {
+				l.Error(err, "Unable to retrive appProjectRepository", appProjectRepository.GetAsLogKeyValues()...)
+
+				// If AppProjectRepository is not present in DB, create it.
+				if db.IsResultNotFoundError(err) {
```

Comment:
Yes, but while I checked it once again I think we don't need to update repoURL in appProjectRepository because clusterUser and RepoURL are unique constraint if RepoURL is updated it is going to create corresponding new appProjectRepo entry in DB rather than updating it, so I think updating RepositorycredentialsID  is enough, Please correct me if I am assuming this wrong ?